src/reanimated2/animation/timing.ts

```diff
@@ -65,6 +69,10 @@ export const withTiming = function (
 ): Animation<TimingAnimation> {
   'worklet';
 
+  if (__DEV__ && userConfig?.easing) {
+    assertEasingIsWorklet(userConfig?.easing);
```

Comment:
You can consider if this additional nullcheck is necessary 
```suggestion
    assertEasingIsWorklet(userConfig.easing);
```